### PR TITLE
WebGL Minimap

### PR DESCRIPTION
**Pull request by ds300**
This PR replaces our current minimap implementation with one that uses WebGL

### Change Type

- [x] `sdk` — Changes the tldraw SDK
- [ ] `dotcom` — Changes the tldraw.com web app
- [ ] `docs` — Changes to the documentation, examples, or templates.
- [ ] `vs code` — Changes to the vscode plugin
- [ ] `internal` — Does not affect user-facing stuff

- [ ] `bugfix` — Bug fix
- [ ] `feature` — New feature
- [x] `improvement` — Improving existing features
- [ ] `chore` — Updating dependencies, other boring stuff
- [ ] `galaxy brain` — Architectural changes
- [ ] `tests` — Changes to any test code
- [ ] `tools` — Changes to infrastructure, CI, internal scripts, debugging tools, etc.
- [ ] `dunno` — I don't know

### Test Plan

1. Add a step-by-step description of how to test your PR here.
2.

- [ ] Unit Tests
- [ ] End to end tests

### Release Notes

- Add a brief release note for your PR here.

### CODE REVIEW

File: packages/editor/src/lib/primitives/geometry/Geometry2d.ts [hunk 1]
```diff
@@ -1,3 +1,6 @@
+import earcut from 'earcut'
+// @ts-expect-error
+import extrudePolyline from 'extrude-polyline'
```
Comment (ds300) [inline packages/editor/src/lib/primitives/geometry/Geometry2d.ts:3]: hm I'm gonna extract this stuff out and only do webgl stuff in the minimap bit

**Comment by steveruizok** [comment]
Quick list of bugs, plz excuse brevity

- checkbox shape cursed
- fast draw shapes look polyliney
- highlighter shape sometimes closes
- cloud shape spikey
- draw shape can be spikey

**Comment by MitjaBezensek** [comment]
Noticed similar things as Steve did. 

When shapes get close to 0 size in one of the dimensions they can bug out. Example with the star:

https://github.com/tldraw/tldraw/assets/2523721/47acc270-03c4-4b45-8987-bb05a3794349

Similar thing when resizing:

https://github.com/tldraw/tldraw/assets/2523721/f4b370e0-8214-4e48-919d-ffecda2d5f45

Maybe we should use some transparency for the highlighter? Would be nice to kinda see through it, the same as on the canvas.

**Comment by ds300** [comment]
I'm gonna put this on ice for a little bit. I'm on holiday next week and I have some other tasks to get done before then and this feels like it's gonna take a good deal more tweaking to get it over the line.

**Pull request merged**